lib/livebook/runtime/attached.ex

```diff
@@ -92,7 +92,7 @@ defmodule Livebook.Runtime.Attached do
     # the requirement we enforce.
 
     current = Version.parse!(System.version())
-    same_minor = "#{current.major}.#{current.minor}.0"
+    same_minor = "#{current.major}.#{current.minor}.0-dev"
```

Comment:
What about this?

```elixir
current = System.version()

same_minor =
  current
  |> Version.parse!()
  |> Map.replace!(:patch, 0)
  |> Version.to_string()
```

This means that if Livebook is running on `-dev`, we include `-dev` in the requirement, otherwise we don't.

If the user is not used to running Elixir main, showing `-dev` in the error message may be unnecessarily confusing.

(Sidenote: note the intentional `current = System.version()`, it's expected to be a string below, currently it is a bug)